pick.lic

```diff
@@ -708,48 +714,104 @@ class Pick
+        when /You'd better tie it up/, /You'll need to tie it up before/
+          #tie pouch once there is some gems in it
+          DRC.bput("tie my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun}", 'You tie', "it's empty?", 'has already been tied off') if @tie_gem_pouches
+          DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{@settings.autoloot_container}", /You open/, /is too full to fit another gem/, /is too full to fit any more gems/, /You'd better tie it up before putting/, /You'll need to tie it up before/)    		
+        end
+    end	  
+				
+    
+    if (@settings.fill_pouch_with_box || @settings.loot_specials.empty?) && !(@settings.fill_pouch_with_box == false) 
       # Don't handle full pouches here, cause if we do, we can't tie them (when they're empty)
-      DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{box_noun}",
-        'You fill your', 'You open your', 'What were you referring to', /any gems/, /too full to fit/)
+      DRC.bput("fill my #{@settings.gem_pouch_adjective} #{@settings.gem_pouch_noun} with my #{box_noun}", 'You fill your', 'You open your', 'What were you referring to', /any gems/, /too full to fit/)
     end
 
-    box_items = DRCI.get_item_list(box_noun, 'look');
-    loot = box_items.map { |item| DRC.get_noun(item) }
+    box_items = DRCI.get_item_list(box_noun, 'look');	
+    echo("boxitems = #{box_items}") if @debug
+    if box_items == 'There is nothing in there\.'
+      echo 'All loot gathered, dumping box'
+      return
+    end
+	
+    #loot each item if there is anything left.
+    loot = box_items.map { |item| item }
     echo "box items: #{box_items} - loot: #{loot}" if @debug
-
     loot.each { |item| loot_item(item, box_noun) }
 
     # Recurse for very full boxes
     loot(box_noun) if loot.last() =~ /stuff/i
+
   end
 
   def loot_item(item, box)
+    	
+    item_long = item
+    item = DRC.get_noun(item)
+	
+    echo("loot_item = #{item}") if @debug
+    echo("item_long = #{item_long}") if @debug
+	
     # We don't touch glowing fragments, since we took off all our armor...
     return if item =~ /fragment/i
     # Stuff happens for very full boxes, and isn't really loot
     return if item =~ /stuff/i
 
-    item_long = nil
-    case DRC.bput("get #{item} from my #{box}", 'You get (.*) from inside', 'You pick up \d* \w* (?:lirum|dokora|kronar)s?')
-    when /You pick up/ # Coins
+    #search for loose gems
+    if @gem_nouns.find { |thing| thing =~ /\b#{item}\b/i }
+      echo "gem found, getting and stowing it"
+        DRC.bput("get #{item} from my #{box}", 'You get (.*) from inside', 'You pick up \d* \w* (?:lirum|dokora|kronar)s?')
+        case DRC.bput("stow my #{item}", /You put/, /You open/, /is too full to fit another gem/, /You'd better tie it up before putting/)
```

Comment:
💡 Could these get/stow be replaced with DRCI methods?
```ruby
DRCI.get_item?(item, box)
DRCI.stow_item?(item)
```

Reply:
i did the first one as suggested for get, tested ok, but if a gem pouch is full, then we need to get another pouch etc, and stow_item does not cover that scenario from what i can see? and if i used that would need to except it with what I have i think anyhow? sorry might be able to be done but not sure.